packages/ui/src/hooks/useQueues.ts

```diff
@@ -0,0 +1,42 @@
+import { useRef } from 'react'
+
+export function useQueues() {
+  const runningTaskRef = useRef<null | Promise<void>>(null)
+  const latestTaskRef = useRef<(() => Promise<void>) | null>(null)
+  const abortControllerRef = useRef<AbortController | null>(null)
+
+  const queueTask = (fn: () => Promise<void>) => {
+    // Overwrite the latest task
+    latestTaskRef.current = fn
+
+    // If there's a running task, don't stack more tasks; just update `latestTaskRef`
+    if (runningTaskRef.current !== null) {
+      return
+    }
+
+    const executeTask = async () => {
+      while (latestTaskRef.current) {
+        const taskToRun = latestTaskRef.current
+        latestTaskRef.current = null // Reset latest task before running
+
+        const controller = new AbortController()
+        abortControllerRef.current = controller
```

Comment:
This abort controller is not used anywhere besides being assigned here nor is it exported. Do we need it?

Reply:
It was unused before, largely because abort controllers in general don't work as expected with server functions. But I'm now passing it through the `taskToRun`.

Reply:
This turned out to be a bad idea, see #11579.

Reply:
This turned out to be a bad idea, see #11867.